### PR TITLE
All Changes to Chirper

### PR DESCRIPTION
**Pull request by dsargent25**

### CODE REVIEW

File: database/migrations/0001_01_01_000000_create_users_table.php [hunk 1]
```diff
@@ -19,6 +19,8 @@ public function up(): void
             $table->string('password');
             $table->rememberToken();
             $table->timestamps();
+            $table->string('profile_image_url');
+            $table->integer('chirp_count');
```
Comment (shepshep7) [inline database/migrations/0001_01_01_000000_create_users_table.php]: remove this column, and do a fresh migration I guess so that you don't have it anymore. you can determine this property by just counting the chirps a user has

File: app/Http/Controllers/ChirpController.php [hunk 2]
```diff
@@ -35,6 +35,10 @@ public function show(Chirp $chirp)
      //   
     }
 
+
+
+//Chirp All Chirpers Method
+
     public function all(User $user): View
```
Comment (shepshep7) [inline app/Http/Controllers/ChirpController.php]: move this to a UserController and rename to index. the index method should get all users, sort by newest, and withCount their chirps so you can have a count of their posts

File: app/Models/User.php [hunk 3]
```diff
@@ -20,7 +20,9 @@ class User extends Authenticatable
     protected $fillable = [
         'name',
         'email',
-        'password'
+        'password',
+        'profile_image_url',
+        'chirp_count'
```
Comment (shepshep7) [inline app/Models/User.php]: remove chirp count

File: database/migrations/2024_08_29_175846_create_comments_table.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('comments', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('chirp_id')->constrained()->cascadeOnDelete();
```
Comment (shepshep7) [inline database/migrations/2024_08_29_175846_create_comments_table.php:16]: good syntax here!

File: resources/views/chirps/all.blade.php
Comment (shepshep7) [inline resources/views/chirps/all.blade.php:1]: <p>{{$user->chirp_count}}</p> on line 43 change this to accept the chirps_count attribute returned by the withCount eloquent

File: resources/views/layouts/navigation.blade.php [hunk 5]
```diff
@@ -21,7 +21,9 @@
                 <x-nav-link :href="route('chirps.latest')" :active="request()->routeIs('chirps.latest')">
                         {{ __('Recent Chirps') }}
                 </x-nav-link>
-
+                <x-nav-link :href="route('chirps.all')" :active="request()->routeIs('chirps.all')">
```
Comment (shepshep7) [inline resources/views/layouts/navigation.blade.php]: change to reflect new usercontroller and index route

File: routes/web.php [hunk 6]
```diff
@@ -25,5 +25,19 @@
 Route::put('chirps/{chirp}', [ChirpController::class, 'update'])->name('chirps.update');
 Route::delete('chirps/{chirp}', [ChirpController::class, 'destroy'])->name('chirps.destroy');
 Route::get('chirps/latest', [ChirpController::class, 'latest'])->name('chirps.latest');
-Route::get('chirps/all', [ChirpController::class, 'all'])->name('chirps.all');
+
+//Route for Displaying All Chirpers
+Route::get('chirps/all', [ChirpController::class, 'all'])->name('chirps.all');
```
Comment (shepshep7) [inline routes/web.php]: change to reflect new UserController and index method

File: resources/views/welcome.blade.php
Comment (shepshep7) [inline resources/views/welcome.blade.php:1]: move the login and register buttons from welcome.blade.php to guest.blade.php. change the create user and login user views to extend the guest layout

File: routes/web.php
Comment (shepshep7) [inline routes/web.php:1]: Route::get('/', function () {
    return view('welcome');
}); starting on line 8

this should either return the users dashboard if they are logged in, or return the guest view if they are not

**Review by shepshep7 — CHANGES_REQUESTED** [review]
Make all of the commented changes, then install tailwindcss and resubmit your pull request

File: app/Http/Controllers/ChirpController.php [hunk 7]
```diff
@@ -62,10 +55,13 @@ public function store(Request $request): RedirectResponse
 
 	$request->user()->chirps()->create($validated);
 
+    $userid = Auth::user()->id;
+    $users = User::find($userid);
```
Comment (shepshep7) [inline app/Http/Controllers/ChirpController.php]: why is this here

File: app/Http/Controllers/ChirpController.php [hunk 8]
```diff
@@ -111,6 +107,9 @@ public function destroy(Chirp $chirp): RedirectResponse
 
         $chirp->delete();
 
+        $userid = Auth::user()->id;
```
Comment (shepshep7) [inline app/Http/Controllers/ChirpController.php]: why is this here

File: app/Http/Controllers/CommentController.php [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Gate;
+
+
+class CommentController extends Controller
+{
+
+
+    public function create(Chirp $chirp)
+    {
```
Comment (shepshep7) [inline app/Http/Controllers/CommentController.php]: delete this method

File: resources/views/chirps/index.recentchirps.php
Comment (shepshep7) [inline resources/views/chirps/index.recentchirps.php:1]: this is not named properly

File: resources/views/layouts/guest.blade.php [hunk 10]
```diff
@@ -15,7 +15,46 @@
...
+
+                
+                @if (Route::has('login'))
+                <nav class="-mx-3 flex flex-1 justify-end"
+                     style="padding:1rem;"
+                        >
+                    @auth
+                        {{-- <a
+                            href="{{ url('/dashboard') }}"
+                            class="rounded-md px-3 py-2 text-black ring-1 ring-transparent transition hover:text-black/70 focus:outline-none focus-visible:ring-[#FF2D20] text-black dark:hover:text-white/80 dark:focus-visible:ring-white"
```
Comment (shepshep7) [inline resources/views/layouts/guest.blade.php]: remove this commented code

File: resources/views/user/user.blade.php
Comment (shepshep7) [inline resources/views/user/user.blade.php:1]: if there isn't a specific reason for this blade remove it

File: resources/views/chirps/index.recentchirps.php
Comment (shepshep7) [inline resources/views/chirps/index.recentchirps.php:1]: change file name

File: resources/views/welcome.blade.php
Comment (shepshep7) [inline resources/views/welcome.blade.php:1]: delete this file

File: resources/views/layouts/navigation.blade.php [hunk 11]
```diff
@@ -12,26 +12,27 @@
...
                 </div>
             </div>
 
             <!-- Settings Dropdown -->
             <div class="hidden sm:flex sm:items-center sm:ms-6">
                 <x-dropdown align="right" width="48">
                     <x-slot name="trigger">
-                        <button class="inline-flex items-center px-3 py-2 border border-transparent text-sm leading-4 font-medium rounded-md text-gray-500 bg-white hover:text-gray-700 focus:outline-none transition ease-in-out duration-150">
+                        <button class="inline-flex items-center px-3 py-2 border border-transparent text-sm leading-4 font-medium rounded-md text-gray-500 bg-white hover:text-gray-700 focus:outline-none transition ease-in-out duration-150" style="background-color:#26A7DE;color:white;">
                             <div>{{ Auth::user()->name }}</div>
```
Comment (shepshep7) [inline resources/views/layouts/navigation.blade.php:36]: add null protection here in case there is no current logged in user

File: app/Policies/ChirpPolicy.php
Comment (shepshep7) [inline app/Policies/ChirpPolicy.php:1]: finish this

File: app/Policies/CommentPolicy.php
Comment (shepshep7) [inline app/Policies/CommentPolicy.php:1]: make sure this works

**Review by shepshep7 — CHANGES_REQUESTED** [review]
looks good, just a few more changes

File: app/Http/Controllers/ChirpController.php (hunk 7 above)
Comment (dsargent25) [reply app/Http/Controllers/ChirpController.php]: This was old logic for incrementing "chirpCount". Removed.

File: app/Http/Controllers/ChirpController.php (hunk 8 above)
Comment (dsargent25) [reply app/Http/Controllers/ChirpController.php]: This was left over from decrementing "chirpCount". Also removed.

File: resources/views/chirps/index.recentchirps.php
Comment (dsargent25) [reply resources/views/chirps/index.recentchirps.php:1]: Renamed to index.recent-chirps.php.

File: resources/views/layouts/navigation.blade.php (hunk 11 above)
Comment (dsargent25) [reply resources/views/layouts/navigation.blade.php:36]: I added logic to navigation.blade.php to change name to "guest" in the case that they can still access said page. Otherwise, all routes now have middleware auth, so if any page is manually entered, they just get redirected to the login page.

File: app/Http/Controllers/ChirpController.php
Comment (dsargent25) [inline app/Http/Controllers/ChirpController.php:1]: Removed leftover logic for chirpCount where I was incrementing/decrementing the value.

File: resources/views/layouts/guest.blade.php
Comment (dsargent25) [inline resources/views/layouts/guest.blade.php:1]: This was meant to keep Register and Login displayed on the home page. Rather than commenting this out, this should be deleted (or properly re-implemented.)

File: routes/web.php
Comment (dsargent25) [inline routes/web.php:1]: Repetition of "chirps/" is redundant in the first route grouping and can be integrated back in. I just did this to 100% make sure I was getting the routes right.

File: app/Models/User.php [hunk 12]
```diff
@@ -3,6 +3,7 @@
 namespace App\Models;
 
 use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\HasManyThrough;
 // use Illuminate\Contracts\Auth\MustVerifyEmail;
```
Comment (shepshep7) [inline app/Models/User.php]: remove

File: app/Providers/AppServiceProvider.php [hunk 13]
```diff
@@ -2,6 +2,9 @@
 
 namespace App\Providers;
 
+use App\Models\Comment;
```
Comment (shepshep7) [inline app/Providers/AppServiceProvider.php]: remove these, no longer used

File: resources/views/chirps/index.recent-chirps.php
Comment (shepshep7) [inline resources/views/chirps/index.recent-chirps.php:1]: either delete this file or change the naming pattern

File: resources/views/layouts/guest.blade.php
Comment (shepshep7) [reply resources/views/layouts/guest.blade.php:1]: I think the guest layout is only used if the user is not authenticated right? you can get rid of those commented route:has blocks, leaving the routes in place

File: routes/web.php [hunk 14]
```diff
@@ -3,13 +3,14 @@
...
 use App\Http\Controllers\CommentController;
 use App\Http\Controllers\ProfileController;
+use App\Http\Controllers\UserController;
 use Illuminate\Support\Facades\Route;
 
 Route::get('/', function () {
-    return view('welcome');
-});
+    return view('guest');
+})->middleware(['guest'])->name('guest');
```
Comment (shepshep7) [inline routes/web.php]: you shouldn't need this route. if your authenticate middleware properly redirects to the login or register page when user not authenticated this isn't necessary

**Review by shepshep7 — CHANGES_REQUESTED** [review]
minor changes requested, otherwise looks good

File: app/Providers/AppServiceProvider.php
Comment (shepshep7) [inline app/Providers/AppServiceProvider.php:1]: leave the register and boot methods

**Review by shepshep7 — APPROVED** [review]
